src/function/scalar/map/map_extract.cpp

```diff
@@ -5,34 +5,76 @@
 #include "duckdb/common/types/data_chunk.hpp"
 
 namespace duckdb {
-void FillResult(Value &values, Vector &result, idx_t row) {
-	//! First Initialize List Vector
-	idx_t current_offset = ListVector::GetListSize(result);
-	//! Push Values to List Vector
-	auto &list_values = ListValue::GetChildren(values);
-	for (idx_t i = 0; i < list_values.size(); i++) {
-		ListVector::PushBack(result, list_values[i]);
+void FillResult(Vector &map, Vector &offsets, Vector &result, idx_t count) {
+	UnifiedVectorFormat map_data;
+	map.ToUnifiedFormat(count, map_data);
+
+	UnifiedVectorFormat offset_data;
+	offsets.ToUnifiedFormat(count, offset_data);
+
+	UnifiedVectorFormat result_data;
+	result.ToUnifiedFormat(count, result_data);
+
+	auto entry_count = ListVector::GetListSize(map);
+	auto &values_entries = MapVector::GetValues(map);
+	UnifiedVectorFormat values_entry_data;
+	// Note: this vector can have a different size than the map
+	values_entries.ToUnifiedFormat(entry_count, values_entry_data);
+
+	for (idx_t row = 0; row < count; row++) {
+		idx_t offset_idx = offset_data.sel->get_index(row);
+		auto offset = ((int32_t *)offset_data.data)[offset_idx];
+
+		if (!offset) {
+			// 'key' was not found in the keys list of the map
+			// FIXME: does this need to have the selection vector applied??
+			result_data.validity.SetInvalid(row);
+
+			// Get the current size of the list, for the offset
+			idx_t current_offset = ListVector::GetListSize(result);
+			// Set the entry data for this result row
+			idx_t result_index = result_data.sel->get_index(row);
+			auto &entry = ((list_entry_t *)result_data.data)[result_index];
+			entry.length = 0;
+			entry.offset = current_offset;
+			continue;
+		}
+		// Get the current size of the list, for the offset
+		idx_t current_offset = ListVector::GetListSize(result);
+
+		// Get the 'values' list entry corresponding to the offset
+		idx_t value_index = map_data.sel->get_index(row);
+		auto &value_list_entry = ((list_entry_t *)map_data.data)[value_index];
+
+		// Add the values to the result
+		idx_t list_offset = value_list_entry.offset + (offset - 1);
+		// All keys are unique, only one will ever match
+		idx_t length = 1;
```

Comment:
Looking back on this, this seems a wrong assumption
but it works?

This passes:
```sql
query I
select map([[[3]]], [[[4],[4]]])[[[3]]]
----
[[[4], [4]]]
```

Reply:
Ah nvm, the input is a LIST, so it realizes that we're copying 1 entry from a list - thereby copying also the contents of that entry